src/Service/ClientService.php

```diff
@@ -62,17 +82,45 @@ public function __construct(
             $liveEndpointUrlPrefix = $this->configurationService->getLiveEndpointUrlPrefix();
 
             $this->setXApiKey($apiKey);
-            $this->setAdyenPaymentSource("Module", "Version"); //TODO fetch data from the plugin
-            $this->setMerchantApplication("Module", "Version"); //TODO fetch data from the plugin
-            $this->setExternalPlatform("Platform", "Version"); //TODO fetch data from the platform
+            $this->setMerchantApplication(self::MERCHANT_APPLICATION_NAME, $this->getModuleVersion());
+            $this->setExternalPlatform(self::EXTERNAL_PLATFORM_NAME, $this->storeService->getShopwareVersion());
             $this->setEnvironment($environment, $liveEndpointUrlPrefix);
 
             $this->setLogger($apiLogger);
-
             //TODO set $this->configuration
         } catch (\Exception $e) {
             $this->genericLogger->error($e->getMessage());
             // TODO: check if $environment is test and, if so, exit with error message
         }
     }
+
+    /**
+     * Get adyen module's version from composer.json
```

Comment:
```suggestion
     * Get adyen module's version from composer.json
     * TODO: switch to a shopware service to retrieve the module version instead of composer
```